### PR TITLE
v20241025 release notes (taking over from Alvin)

### PR DESCRIPTION
**Pull request by phealy**
New release notes PR, taking over for Alvin.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,57 @@
 # Azure Kubernetes Service Changelog
 
+## Release 2024-10-25
+
+Monitor the release status by regions at [AKS-Release-Tracker](https://releases.aks.azure.com/). This release is titled as `v20241025`.
+
+### Announcements
+
+* AKS version 1.28 End of Life is Jan, 15 2025.
+* AKS will be upgrading the KEDA addon to more recent [KEDA versions](https://github.com/kedacore/keda/releases). The AKS team has added KEDA 2.15 on AKS clusters with K8s versions >=1.31, KEDA 2.14 for Kubernetes v1.30. KEDA 2.15 and KEDA 2.14 will introduce multiple breaking changes. View the [troubleshooting guide](https://learn.microsoft.com/troubleshoot/azure/azure-kubernetes/extensions/changes-in-kubernetes-event-driven-autoscaling-add-on-214-215) to learn how to mitigate these breaking changes.
```
Comment (qpetraroia) [inline CHANGELOG.md:10]: AKS will be upgrading the KEDA addon to more recent [KEDA versions](https://github.com/kedacore/keda/releases). The AKS team has added KEDA 2.15 on AKS clusters with K8s versions >=1.32. 

Can we change it to 1.32? thanks

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,57 @@
...
+* As of November 30th, 2024, new AKS clusters created with Kubernetes versions 1.28 and 1.29 will no longer support beta Kubernetes APIs.
+
+### Release Notes
+
+* Features:
+  * AKS patch versions 1.28.14, 1.29.9, 1.30.5 are now available. Refer to [version support policy](https://learn.microsoft.com/azure/aks/supported-kubernetes-versions?tabs=azure-cli#kubernetes-version-support-policy) and [upgrading a cluster](https://learn.microsoft.com/azure/aks/upgrade-aks-cluster?tabs=azure-cli) for more information.
+  * AKS version [`1.31`](https://kubernetes.io/blog/2024/08/13/kubernetes-v1-31-release/) is now generally available. Please check the release tracker for when your region will receive the GA update. Some regions may not receive this update until later in November.
+  * The first official patch version of AKS LTS 1.27, 1.27.100, is being released.
+ 
+* Preview features:
```
Comment (qpetraroia) [inline CHANGELOG.md:24]: Can we add 

[GitHub Copilot](https://azure.microsoft.com/en-us/products/github/Copilot?msockid=39613fded86166a01c612b53d9db67d8) for Azure now supports [AKS commands](https://learn.microsoft.com/en-us/azure/aks/aks-extension-ghcopilot-plugins).

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,57 @@
...
+    * [Azure Disk CSI Driver](https://github.com/kubernetes-sigs/azuredisk-csi-driver/releases): v1.28.11 for AKS 1.27, v1.19.10 for AKS 1.28 and 1.29, and v1.30.5 for AKS 1.30+
+    * [Azure Files CSI Driver](https://github.com/kubernetes-sigs/azurefile-csi-driver/releases): v1.28.13 for AKS 1.27, v1.29.9 for AKS 1.28, v1.30.6 for AKS 1.29+
+  * Azure Monitor for Containers has been upgraded to [3.1.24](https://github.com/microsoft/Docker-Provider/blob/ci_prod/ReleaseNotes.md#10072024--).
+  * AKS Windows Server 2019 image has been updated to [AKSWindows-2019-17763.6414.241010](vhd-notes/AKSWindows/2019/17763.6414.241010.txt).
+  * AKS Windows Server 2022 image has been updated to [AKSWindows-20348.2762.241009](vhd-notes/AKSWindows/2022/20348.2700.240911.txt).
+  * AKS Azure Linux image has been updated to [202410.27.0](vhd-notes/AzureLinux/202410.27.0.txt).
+  * AKS Ubuntu image has been updated to [202410.27.0](vhd-notes/aks-ubuntu/202410.27.0.txt).
+  * cost-analysis-agent image has been updated to v0.0.18
+  * ip-masq-agent image has been updated to [v0.1.14](https://github.com/Azure/ip-masq-agent-v2/releases/tag/v0.1.14)
+
```
Comment (haitch) [inline CHANGELOG.md:54]: - AKS runCommand, kubectl to 1.30.5, helm to 3.15.4, also added couple new binary: jq, awk, grep, xargs, fix https://github.com/Azure/AKS/issues/3034

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,57 @@
...
+* Starting on January 1, 2025, [invalid values sent to the Azure AKS API for the properties.mode field of AKS AgentPools will be rejected](https://github.com/Azure/AKS/issues/4468). Prior to this change, unknown modes were assumed to be User. The only valid values for this field are the (case-sensitive) strings:["User", "System"](https://learn.microsoft.com/rest/api/aks/agent-pools/create-or-update?view=rest-aks-2024-02-01&tabs=HTTP#agentpoolmode), or ["Gateway"](https://learn.microsoft.com/rest/api/aks/agent-pools/create-or-update?view=rest-aks-2024-06-02-preview&tabs=HTTP#agentpoolmode).
+* AKS will start to block new cluster creation with basic load balancer in January 2025. Basic Load Balancer will be deprecated September 31 2025 and affected clusters must be migrated to the Standard Load Balancer prior to that date. Refer to [BLB deprecation announcement](
+https://azure.microsoft.com/updates/azure-basic-load-balancer-will-be-retired-on-30-september-2025-upgrade-to-standard-load-balancer/) for more information.
+* As of November 30th, 2024, new AKS clusters created with Kubernetes versions 1.28 and 1.29 will no longer support beta Kubernetes APIs.
```
Comment (shashankbarsin) [inline CHANGELOG.md:15]: what about 1.30+?

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,57 @@
...
+  * AKS version [`1.31`](https://kubernetes.io/blog/2024/08/13/kubernetes-v1-31-release/) is now generally available. Please check the release tracker for when your region will receive the GA update. Some regions may not receive this update until later in November.
+  * The first official patch version of AKS LTS 1.27, 1.27.100, is being released.
+ 
+* Preview features:
+  * We've added [a new way to optimize your upgrade process drain behavior](https://learn.microsoft.com/en-us/azure/aks/upgrade-cluster?tabs=azure-cli#optimize-for-undrainable-node-behavior-preview). By default, a node drain failure causes the upgrade operation to fail, leaving the undrained nodes in a schedulable state, this behavior is called `Schedule`. Alternatively, you can select the `Cordon` behavior, which skips nodes that fail to drain by placing them in a quarantined state, labeling them `kubernetes.azure.com/upgrade-status:Quarantined` and proceeds with upgrading the remaining nodes. This ensures that all nodes are either upgraded or quarantined. This approach allows you to troubleshoot drain failures and gracefully manage the quarantined nodes.
```
Comment (shashankbarsin) [inline CHANGELOG.md:25]:
```suggestion
  * We've added [a new way to optimize your upgrade process drain behavior](https://learn.microsoft.com/azure/aks/upgrade-cluster?tabs=azure-cli#optimize-for-undrainable-node-behavior-preview). By default, a node drain failure causes the upgrade operation to fail, leaving the undrained nodes in a schedulable state, this behavior is called `Schedule`. Alternatively, you can select the `Cordon` behavior, which skips nodes that fail to drain by placing them in a quarantined state, labeling them `kubernetes.azure.com/upgrade-status:Quarantined` and proceeds with upgrading the remaining nodes. This ensures that all nodes are either upgraded or quarantined. This approach allows you to troubleshoot drain failures and gracefully manage the quarantined nodes.
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,57 @@
...
+  * You can now [block pod access to the Azure Instance Metadata Service (IMDS) endpoint](https://learn.microsoft.com/azure/aks/imds-restriction) to enhance security.
+  * Azure Linux v3 is now in preview for AKS 1.31 clusters. After registering the preview flag `AzureLinuxV3Preview` newly created AzureLinux node pools will receive the v3 image. Existing Azure Linux v2 node pools will not upgrade to v3 and must be recreated to upgrade.
+    * NOTE: Azure Linux v3 changes the cryptographic provider to OpenSSL + SymCrypt. The SymCrypt library will operate in FIPS mode but is still in the final stages of the validation process and thus **is not considered to be FIPS-validated at this time**. Do not use this preview with FIPS-enabled node pools if you must use a FIPS-validated cryptographic library.
```
Comment (shashankbarsin) [inline CHANGELOG.md:28]: nit: indentation is off, has to be reduced

Comment (shashankbarsin) [reply CHANGELOG.md:28]: have we announced Azure Linux v3 preview before?

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,57 @@
...
+
+* Behavior change:
+  * You can now fine-tune supported models on KAITO version [0.3.1](https://github.com/Azure/kaito/releases/tag/v0.3.1) with the AI toolchain operator add-on on your AKS cluster.
```
Comment (shashankbarsin) [inline CHANGELOG.md:31]: is there a behavior change for models already deployed? or is this fine-tune customization a new capability? if latter, should we move this under preview features section and call it as an enhancement?

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,57 @@
...
+  * You can now skip one release [while upgrading Azure Service Mesh](https://learn.microsoft.com/azure/aks/istio-upgrade) as long as the destination release is a supported revision - for example, asm-1-21 can upgrade directly to asm-1-23.
```
Comment (shashankbarsin) [inline CHANGELOG.md:32]: this is not a behavior change as such. users who were doing upgrade from N to N+1 won't be impacted and this only introduces an option to skip the release. this line can be moved under preview features section

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,57 @@
...
+  * Virtual Machine nodepools will not be able to use custom VNets when if the cluster is using system-assigned identity.
```
Comment (shashankbarsin) [inline CHANGELOG.md:33]:
```suggestion
  * Virtual Machine node pools will not be able to use custom virtual networks (vnets) if the cluster is using system-assigned identity.
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,57 @@
...
+  * LTS clusters no longer perform defaulting on autoUpgrader profile.
```
Comment (shashankbarsin) [inline CHANGELOG.md:34]: What was the default before within the profile? Can we add more details?

File: CHANGELOG.md (hunk 4 above)
Comment (phealy) [reply CHANGELOG.md:15]: That's already disabled in 1.30+, so this is rolling the change farther back.

File: CHANGELOG.md (hunk 8 above)
Comment (phealy) [reply CHANGELOG.md:32]: Is it really a preview feature - is there a feature flag or something that has to be registered, and is it fully supported? I would tend to class this more as a behavior change, but maybe announcements would be better?

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,5 +1,57 @@
...
+  * AKS CoreDNS configuration will now block all queries ending in `reddog.microsoft.com` and some queries ending in `internal.cloudapp.net` from being forwarded to upstream DNS when they are the result of improper search domain completion. See [the documentation for more details.](https://learn.microsoft.com/azure/aks/coredns-custom#invalid-search-domain-completions-for-internalcloudappnet-and-reddogmicrosoftcom)
+  * Azure NPM's CPU request has been lowered from 250m to 50m.
+  * Azure CNI Overlay now checks that the pod CIDR does not conflict with any subnet in the VNet, rather than checking if it conflicts with the VNet address space as a whole.
+
+* Component updates:
+  * The KEDA addon will be bumped to version 1.15 on AKS 1.32 instead of AKS 1.31.
```
[the remark this answers is not in the document]
Comment (phealy) [reply CHANGELOG.md:40]: Removing this as the announcement was changed to reflect the new version.

File: CHANGELOG.md (hunk 5 above)
Comment (phealy) [reply CHANGELOG.md:25]: good catch, fixed

File: CHANGELOG.md (hunk 6 above)
Comment (shashankbarsin) [reply CHANGELOG.md:28]: got it. can we add a docs link for Azure Linux v3 preview announcement, rest looks good.

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,5 +1,59 @@
...
+    * NOTE: Azure Linux v3 changes the cryptographic provider to OpenSSL + SymCrypt. The SymCrypt library will operate in FIPS mode but is still in the final stages of the validation process and thus **is not considered to be FIPS-validated at this time**. Do not use this preview with FIPS-enabled node pools if you must use a FIPS-validated cryptographic library.
+
+* Behavior change:
+  * You can now fine-tune supported models on KAITO version [0.3.1](https://github.com/Azure/kaito/releases/tag/v0.3.1) with the AI toolchain operator add-on on your AKS cluster.
+  * You can now skip one release [while upgrading Azure Service Mesh](https://learn.microsoft.com/azure/aks/istio-upgrade) as long as the destination release is a supported revision - for example, asm-1-21 can upgrade directly to asm-1-23.
+  * Virtual Machine node pools will not be able to use custom virtual networks when if the cluster is using system-assigned identity.
```
Comment (floriankoch) [inline CHANGELOG.md:33]: Can you clarify what this means? Are VMSS also affected?

Comment (phealy) [reply CHANGELOG.md:33]: clarified and updated.

File: CHANGELOG.md (hunk 4 above)
Comment (phealy) [reply CHANGELOG.md]: Done.

File: CHANGELOG.md (hunk 6 above)
Comment (phealy) [reply CHANGELOG.md:30]: For now the link is just to the azure linux docs, as they haven't published their doc PR yet.

File: CHANGELOG.md (hunk 10 above)
Comment (phealy) [reply CHANGELOG.md]: Done. Previously turning on LTS would set the cluster upgrade channel to patch automatically, now it leaves it at the regular default of not enabling auto upgrade.

File: CHANGELOG.md (hunk 7 above)
Comment (phealy) [reply CHANGELOG.md]: moved to features.

**Review by shashankbarsin — APPROVED** [review]

**Review by justindavies — APPROVED** [review]

**Pull request merged**